src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/spell/shape/Zone.java

```diff
@@ -21,8 +23,8 @@ public Zone() {
     public SpellCastResult invoke(ISpell spell, LivingEntity caster, Level level, List<ISpellModifier> modifiers, @Nullable HitResult hit, int ticksUsed, int index, boolean awardXp) {
         if (!level.isClientSide()) {
             com.github.minecraftschurlimods.arsmagicalegacy.common.entity.Zone zone = com.github.minecraftschurlimods.arsmagicalegacy.common.entity.Zone.create(level);
-            zone.setPos(caster.getX(), caster.getY(), caster.getZ());
-            zone.setDeltaMovement(caster.getDeltaMovement());
+            zone.moveTo(hit.getLocation().x(), hit.getLocation().y(), hit.getLocation().z());
+            zone.setDeltaMovement(hit instanceof EntityHitResult entityHit ? entityHit.getEntity().getDeltaMovement() : Vec3.ZERO);
```

Comment:
did zone always move like wave?
if yes it should also fallback to the caster look vector
if no this line makes no sense